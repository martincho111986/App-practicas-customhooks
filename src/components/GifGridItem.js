import React from "react";
import PropTypes from 'prop-types';

const GifGridItem = ({ url, title }) => {
  //  console.log({ id, url, title })

  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p> {title} </p>
    </div>
  );
};

GifGridItem.propTypes ={
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

export default GifGridItem;
