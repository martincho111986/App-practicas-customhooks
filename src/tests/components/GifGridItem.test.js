import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('prueba <GifGridItem />', () => {
    const title = 'Un t√≠tulo';
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('debe mostar el componente <GifGridItem /> correctamente', () => {


        expect(wrapper).toMatchSnapshot()
    })
    test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log(img.props());
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })

    test('debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className')

        expect(className.includes('animate__fadeIn')).toBe( true )
    })
    
    
    
    




})
