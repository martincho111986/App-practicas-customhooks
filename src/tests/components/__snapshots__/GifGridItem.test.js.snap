// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba <GifGridItem /> debe mostar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un título
     
  </p>
</div>
`;
